drivers/sensor/adi/adxl345/adxl345.c

```diff
@@ -519,11 +519,17 @@ static int adxl345_init(const struct device *dev)
 #define ADXL345_CFG_IRQ(inst)
 #endif /* CONFIG_ADXL345_TRIGGER */
 
+#ifdef CONFIG_SPI_RTIO
 #define ADXL345_RTIO_DEFINE(inst)                                    \
 	SPI_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst),     \
 						SPI_WORD_SET(8) | SPI_TRANSFER_MSB |    \
 						SPI_MODE_CPOL | SPI_MODE_CPHA, 0U);     \
 	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
+#elif CONFIG_I2C_RTIO
+#define ADXL345_RTIO_DEFINE(inst)                                    \
+	I2C_DT_IODEV_DEFINE(adxl345_iodev_##inst, DT_DRV_INST(inst));	\
+	RTIO_DEFINE(adxl345_rtio_ctx_##inst, 64, 64);
+#endif
```

Comment:
This won't work if there is one instance of the sensor on an I2C bus and another instance on a SPI bus